lib/registration.pm

```diff
@@ -786,12 +786,10 @@ sub install_docker_when_needed {
     if (is_caasp) {
         # Docker should be pre-installed in MicroOS
         die 'Docker is not pre-installed.' if zypper_call('se -x --provides -i docker');
-    }
-    else {
+    } else {
         if (is_sle('<15')) {
-            assert_script_run('zypper se docker || zypper -n ar -f http://download.suse.de/ibs/SUSE:/SLE-12:/Update/standard/SUSE:SLE-12:Update.repo', timeout => 600);
-        }
-        elsif (is_sle) {
+            add_suseconnect_product('sle-module-containers', 12);
```

Comment:
can we extract this change from this PR. For me it looks generic improvement and this PR already quite big?

Reply:
You're right.